src/key_info_managers/mod.rs

```diff
@@ -50,29 +56,43 @@ struct KeyInfo {
     attributes: Attributes,
 }
 
-impl KeyTriple {
-    /// Creates a new instance of KeyTriple.
-    pub fn new(app_name: ApplicationName, provider_id: ProviderId, key_name: String) -> KeyTriple {
-        KeyTriple {
-            app_name,
-            provider_id,
+impl KeyIdentity {
+    /// Creates a new instance of KeyIdentity.
+    pub fn new(
+        application: ApplicationIdentity,
+        provider: ProviderIdentity,
+        key_name: String,
+    ) -> KeyIdentity {
+        KeyIdentity {
+            application,
+            provider,
             key_name,
         }
     }
 
     /// Checks if this key belongs to a specific provider.
-    pub fn belongs_to_provider(&self, provider_id: ProviderId) -> bool {
-        self.provider_id == provider_id
+    pub fn belongs_to_provider(&self, provider_name: String) -> bool {
```

Comment:
Probably `&str` is better here. Also do we then say that provider names are unique (I think that's fine)?